src/Payum/Action/CaptureAction.php

```diff
@@ -31,10 +31,22 @@ public function execute($request): void
 
         $model = ArrayObject::ensureArrayObject($request->getModel());
 
-        if ($model->offsetExists('payment_state') || $model->offsetExists('error')) {
-            return;
+        if (!$model->offsetExists('payment_state') && !$model->offsetExists('error')) {
+            $this->updateModelForCapture($model);
         }
 
+        if ($model->offsetExists('payment_link') && '' !== $model->get('payment_link')) {
+            throw new HttpRedirect($model->get('payment_link'));
+        }
```

Comment:
If I understand it well the error message is only present if their is an error right ?
So maybe something like this could be enough. 

```suggestion
        if ($model->offsetExists('error')) {
            return;
        }
        
        $this->updateModelForCapture($model);
        
        if ($model->offsetExists('error')) {
            return;
        }
        
        throw new HttpRedirect($model->get('payment_link'));
```

Reply:
No, this will call `updateModelForCapture()` unless an error is already logged. But that method calls the API and will trigger that error if called for a second time.

So my first condition simply checks whether a reponse has already been logged (be it an error or success response). It only calls `updateModelForCapture()` if no known response is logged.

Then the second condition simply makes sure that a `payment_link` is actually set before attempting a redirect.

Reply:
Ok, I see